docs/authoring.md

```diff
@@ -343,4 +343,21 @@ You can use the filter function in your template as in the following example:
 const {{ channelName | camelCase }} = '{{ channelName }}';
 ```
 
+The generator also supports asynchronous filters. Asynchronous filters receive as last argument a callback to resume rendering. Asynchronous filters must be annotated with the `async` keyword. Make sure to call the callback with two arguments: `callback(err, res)`. `err` can be `null`. See the following example of how to use asynchronous filters:
+
```

Comment:
Maybe we should add the fact that promise cannot be used 🤔

Reply:
Or make this workaround https://github.com/asyncapi/generator/pull/516#discussion_r585493590 for Promises 😄

Reply:
I feel like it is more a bandage and complicates the filters more than we should.